### ISSUE TITLE
please help me to fix this bug,How can i do?

### ISSUE THREAD

**Issue by pahnpiseth** [VoiSmart/pjsip-android#31]
libc: ../src/pj/os_core_unix.c:692: pj_thread_this: assertion "!"Calling pjlib from unknown/external thread. You must " "register external threads with pj_thread_register() " "before calling any pjlib functions."" failed

**Comment by FongMi** [comment]
![image](https://user-images.githubusercontent.com/3471963/38123906-8e38abdc-3410-11e8-95dd-b9060a67ec55.png)

**Comment by FongMi** [comment]
I think your sip number is null or empty.

**Comment by pahnpiseth** [comment]
libc: ../src/pj/os_core_unix.c:692: pj_thread_this: assertion "!"Calling pjlib from unknown/external thread. You must " "register external threads with pj_thread_register() " "before calling any pjlib functions."" failed

sometime it crush when hangup or decline,i don't know about.please help...

**Comment by pahnpiseth** [comment]
and sure sip number not null

**Comment by FongMi** [comment]
Can you post your register code ?

**Comment by FongMi** [comment]
Do you have to call to register when calling or decline or hangup?

**Comment by aenonGit** [comment]
Nope

**Comment by zackbugti** [comment]
[Help me urgently, its so urgent please please please]
@FongMi Sir please help me out too. I have downloaded this whole package and built, but it isn't running and I am facing difficulty understanding the code. So please help me out how to make an application based on this. I need to develop just a simple application that could send/receive messages and calls. Please help me.

**Comment by jackkum** [comment]
Hello!

We have the same throuble only android 11 (the other versions works correctly). 
The first call is being successful, but next calls will crash app with the message below
```
A/libc: ../src/pj/os_core_unix.c:692: pj_thread_t *pj_thread_this(void): assertion "!"Calling pjlib from unknown/external thread. You must " "register external threads with pj_thread_register() " "before calling any pjlib functions."" failed
A/libc: Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 27153 (SipService), pid 1254
```

[here is the place of message](https://github.com/pjsip/pjproject/blob/ea7105c222d657d3c1f7fb18ff9130450e2e40e6/pjlib/src/pj/os_core_unix.c#L692)

**Comment by aenonGit** [comment]
have you tried removing this -epConfig.getMedConfig().setThreadCnt(2); `epConfig.getMedConfig().setThreadCnt(2);` ?

**Comment by jackkum** [comment]
@aenonGit 
It doesn't help

**Comment by hchauhansf** [comment]
@Xing1P @jackkum @zackbugti @aenonGit @FongMi 

I am facing the same issue with Android 12 and 13 devices.
Any solution or feedback appreciated.

Thank you.

**Comment by kazimdsaidul** [comment]
same issus

Calling pjlib from unknown/external thread. You must " "register external threads with pj_thread_register() " "before calling any pjlib functions."" failed
2024-11-10 13:18:59.696 10038-10048 libc                                            A  Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 10048 (FinalizerDaemon), pid 10038 ()

**Comment by hchauhansf** [comment]
I have came to conclusion that, this error is more like a generic statement for most of the errors occurs due to the application level code.
My scenario was that I was using a work manager and foreground service which was calling some of the operations for my SDK where I was facing the issue most of the time.
Hence, what I have done is I have used the Handler(Looper.getMainLooper()) for most of the sdk level task to ensure the most of my work happens on the main looper thread only.
